### ISSUE TITLE
Selector for scratch free space in jobs header

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#186]
In GitLab by @mcastril on Sep 13, 2016, 16:13

Today we have problems running some REMOTE_SETUP jobs. According to support there was a job in one specific host using 100% of scratch/temp disk space. It seems that all of our attempts of running REMOTE_SETUP arrived in that same node and crashing in the compilation process.

Operations recommended using this specification chain in the job header to avoid this kind of situations:

#BSUB -R "select[(scratch<90)]"

This would ensure scratch is at 90% or less of usage, so there would be a 10% free.

@dmanubens @ jlope2 @ macosta

**Comment by kinow** [comment]
In GitLab by @jordicuadrado on Sep 14, 2016, 09:58

Hi all, same problem here running auto-nmmb yesterday. Operations recommended the same directive:

*Le recomendamos que incluyera la siguiente directiva de LSF para asegurar que haya un 90% del disco disponible en nodo donde se vaya a ejecutar:*
``#BSUB -R "select[(scratch<10)]"``
*El valor quiere decir que haya hasta un 10% de uso del disco.*

It would be a nice feature if we can select the % value (for a particular job) because in most cases a mere 10% free wouldn't suffice for a successful compilation.

Thanks!

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 14, 2016, 16:13

Do you think that you would need up to 90% of a node storage for a compilation?

**Comment by kinow** [comment]
In GitLab by @jordicuadrado on Sep 14, 2016, 16:20

Hopefully not and not really @mcastril. I wanted to say that it would be nice if it is not a fixed value.
  
As for MN3, I double-checked it and I stand corrected, currently, as stated in https://www.bsc.es/support/MareNostrum3-ug.pdf section 4.3: "Every node has a local hard drive (...) The amount of space within the /scratch filesystem is about 500 GB" so 90% for a compilation won't be needed :)

Thanks!

**Comment by kinow** [comment]
In GitLab by @ jlope2 on Sep 14, 2016, 16:23

I think that there's a confusion with the usage of this directive.

Anyway, the Autosubmit implementation will use the value defined by the user. 

So it's the same if the value means the available space or the used space.

Thanks!

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 14, 2016, 16:38

Thanks @jordicuadrado I tried a quicksearch to look for the aumount of storage in MN3 nodes and I didn't find it.

Joan I think there's no confusion, I asked to support for a confirmation anyway, and they said (I will transcript full conversation):

````
Podéis usar la siguiente directiva para que LSF asigne a vuestro job un nodo que tenga al menos el 10% disponible de espacio en scratch:
#BSUB -R "select[(scratch<10)]"
````
````
Disculpa,

la directiva:
#BSUB -R "select[(scratch<10)]"

Te asegura que como mucho haya un 10% de disco usado, es decir, 90% disponible.
````
````
correcto, si pones <90 aseguras un 10% libre.
Pero es un tanto lioso ese parámetro: si en lugar de la partición scratch se tratara de /tmp (usada por el sistema) entonces sí que se especifica en MB y no en porcentaje.
````

**Comment by kinow** [comment]
In GitLab by @ jlope2 on Sep 14, 2016, 16:44

Okay, so you're right.

It was my fault, I thought @jordicuadrado said that the usage was the opposite way.

Thanks!

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 14, 2016, 16:44

Well, I admit is a little bit tricky, but given that 10% is around 50GB, we can be more or less sure that having a 10% free would allow the compilation to complete.

One thing that just now has come to my mind is: What it would happen if you have free storage when you begin your execution and another process occupies that space while you are already running the compilation? So maybe it's a good idea to give users the power to define that value, cause you have to find a balance between ensuring that you will have enough space and not being so much time in the queues because of asking for too much.

**Comment by kinow** [comment]
In GitLab by @jordicuadrado on Sep 14, 2016, 16:50

Sorry @ jlope2 I wasn't clear, I was just wondering about a user-defined variable for the reasons @mcastril is pointing out and for the (not-so-much probable) case of needing it for another platform.
  
But yes, we were saying the same (Miguel about 90% used (10% free) and I about 90% free (10% used)).

**Comment by kinow** [comment]
In GitLab by @ jlope2 on Sep 15, 2016, 11:08

Status changed to closed by commit 13a347ea2e86a7a539b39bae90011768b5b1e295
